app/views/directives/create-secret.html

```diff
@@ -47,268 +47,296 @@
-        <div class="help-block">
-          Image registry server address is required.
+        <div class="has-error" ng-show="secretForm.passwordToken.$error.required && secretForm.passwordToken.$touched">
+          <div class="help-block">
+            Password or token is required.
+          </div>
+        </div>
+        <div class="help-block" id="password-token-help">
+          Password or token for Git authentication. Required if a ca.crt or .gitconfig file is not specified.
         </div>
-      </div>
 
-      <div class="form-group" ng-class="{ 'has-error' : secretForm.dockerUsername.$invalid && secretForm.dockerUsername.$touched }">
-        <label for="dockerUsername" class="required">Username</label>
-        <div>
-          <input class="form-control"
-            id="dockerUsername"
-            name="dockerUsername"
-            ng-model="newSecret.data.dockerUsername"
-            type="text"
-            autocorrect="off"
-            autocapitalize="none"
-            spellcheck="false"
-            required>
+        <div class="form-group">
+          <div class="checkbox">
+            <label>
+              <input type="checkbox" ng-model="add.cacert">
+              Use a custom ca.crt file
+            </label>
+          </div>
         </div>
-      </div>
-      <div ng-show="secretForm.dockerUsername.$error.required && secretForm.dockerUsername.$touched" class="has-error">
-        <div class="help-block">
-          Username is required.
+        <div class="form-group" ng-if="add.cacert" id="cacert">
+          <label class="required" for="cacert">CA Certificate File</label>
+          <osc-file-input
+            id="cacert-file-input"
+            model="newSecret.data.cacert"
+            drop-zone-id="cacert"
+            help-text="Upload your ca.crt file."
+            required="true"></osc-file-input>
+          <div ui-ace="{
+            mode: 'txt',
+            theme: 'eclipse',
+            rendererOptions: {
+              fadeFoldWidgets: true,
+              showPrintMargin: false
+            }
+          }" ng-model="newSecret.data.cacert" class="create-secret-editor ace-bordered" id="cacert-editor" required></div>
         </div>
+
+
       </div>
 
-      <div class="form-group" ng-class="{ 'has-error' : secretForm.dockerPassword.$invalid && secretForm.dockerPassword.$touched }">
-        <label for="dockerPassword" class="required">Password</label>
-        <div>
-          <input class="form-control"
-            id="dockerPassword"
-            name="dockerPassword"
-            ng-model="newSecret.data.dockerPassword"
-            type="password"
-            autocorrect="off"
-            autocapitalize="none"
-            spellcheck="false"
-            required>
+      <div ng-if="newSecret.authType === 'kubernetes.io/ssh-auth'">
+        <div class="form-group" id="private-key">
+          <label for="privateKey" class="required">SSH Private Key</label>
```

Comment:
for attribute value does not match id value of input

Reply:
It is... under the `label` tag there is a `osc-file-input` directive to which we pass appropriate id in `drop-zone-id`